client/state/data-layer/wpcom-http/utils.js

```diff
@@ -158,3 +163,73 @@ export const dispatchRequest = ( initiator, onSuccess, onError, options ) => ( s
 
 	return initiator( store, action );
 };
+
+/*
+ * Converts an application-level Calypso action that's handled by the data-layer middleware
+ * into a low-level action. For example, HTTP request that's being initiated, or a response
+ * action with a `meta.dataLayer` property.
+ */
+const createRequestAction = ( options, action ) => {
+	const {
+		initiate = noop,
+		onSuccess = noop,
+		onError = noop,
+		onProgress = noop,
+		fromApi = identity,
+	} = options;
+
+	const data = getData( action );
+	if ( data ) {
+		try {
+			return onSuccess( action, fromApi( data ) );
+		} catch ( err ) {
+			return onError( action, err );
+		}
+	}
+
+	const error = getError( action );
+	if ( error ) {
+		return onError( action, error );
+	}
+
+	const progress = getProgress( action );
+	if ( progress ) {
+		return onProgress( action, progress );
+	}
+
+	const initiator = initiate( action );
+	if ( initiator ) {
```

Comment:
So, the desired implicit behavior if `initiate( action )` returns a falsy value is to just silently do nothing? That seems strange to me if someone accidentally passes a function that doesn't have a return or returns null.

Reply:
Oh, that's a good catch! I pushed a commit where a warning is issued when `initiator` doesn't return anything.